server/api/analyzeVotes.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Moderator review of submitted content
+ */
+
+const db = require('../db');
+const { createLogger } = require('../logger');
+
+const logger = createLogger('API/ANALYZE_VOTES');
+
+exports.get = async ({ params: { id } }, res) => {
+  try {
+    const votes = await db.select(
+      `SELECT entry_id, username, rating
+      FROM votes
+      WHERE contest_id = $1`,
+      [id],
+    );
+
+    const userAvg = await db.select(
+      `SELECT username, AVG(rating) as average
+      FROM votes
+      WHERE contest_id = $1
+      GROUP BY username
+      ORDER BY username ASC`,
+      [id],
+    );
+
+    const entryAvg = await db.select(
+      `SELECT entry_id, AVG(rating) as average
+      FROM votes
+      WHERE contest_id = $1
+      GROUP BY entry_id
+      ORDER BY average ASC`,
+      [id],
+    );
+
+    const userEntries = await db.select(
+      `SELECT id, "user"
+      FROM entries e
+      JOIN contest_entries ce
+        ON e.id = ce.entry_id
+      WHERE contest_id = $1`,
+      [id],
+    );
+
+    if (!votes || !userAvg || !entryAvg || !userEntries) {
```

Comment:
I'm not 100% sure, but I believe if the query fails to find anything it will return an empty array, so this condition might never be true? Safest thing might be `if (!votes?.length)`.

Reply:
I think you are right, but it looks like we have a lot of inconsistency in the `server` folder? I see a lot of `if (!contest)` for example, then you have things like `if (!voteDates || !voteDates.length)`

Reply:
Gotcha, thanks for double checking.